lib/geocoding.rb

```diff
@@ -15,13 +15,37 @@ def coordinates
     rescue Geocoder::OverQueryLimitError
       Rails.logger.error("Google Geocoding API responded with OVER_QUERY_LIMIT")
       Geocoder.coordinates(location, lookup: :uk_ordnance_survey_names)
-    end || no_match
+    end || no_coordinates_match
+  end
+
+  def postcode_from_coordinates
+    return Geocoder::DEFAULT_LOCATION if Rails.application.config.geocoder_lookup == :test
+
+    Rails.cache.fetch([:geocoding, location], expires_in: CACHE_DURATION, skip_nil: true) do
```

Comment:
This needs to be a different key from the one on line 13 otherwise it'll overwrite the other cache

Reply:
V.g. point